system-addon/content-src/components/Search/Search.jsx

```diff
@@ -8,33 +8,43 @@ const {actionCreators: ac} = require("common/Actions.jsm");
 class Search extends React.Component {
   constructor(props) {
     super(props);
+    this.controller = null;
     this.onClick = this.onClick.bind(this);
-    this.onInputMount = this.onInputMount.bind(this);
+    this.onFocus = this.onFocus.bind(this);
   }
-
   handleEvent(event) {
     // Also track search events with our own telemetry
     if (event.detail.type === "Search") {
       this.props.dispatch(ac.UserEvent({event: "SEARCH"}));
     }
   }
+  initializeSearch() {
+    if (this.controller) {
+      return;
+    }
+    this.controller = new ContentSearchUIController(this.input, this.input.parentNode,
+      "newtab", "newtab");
+  }
   onClick(event) {
+    if (!this.controller) {
+      this.initializeSearch();
+    }
     this.controller.search(event);
   }
-  onInputMount(input) {
-    if (input) {
-      // The first "newtab" parameter here is called the "healthReportKey" and needs
-      // to be "newtab" so that BrowserUsageTelemetry.jsm knows to handle events with
-      // this name, and can add the appropriate telemetry probes for search. Without the
-      // correct name, certain tests like browser_UsageTelemetry_content.js will fail (See
-      // github ticket #2348 for more details)
-      this.controller = new ContentSearchUIController(input, input.parentNode,
-        "newtab", "newtab");
-      addEventListener("ContentSearchClient", this);
-    } else {
-      this.controller = null;
-      removeEventListener("ContentSearchClient", this);
-    }
+  onFocus(e) {
+    this.initializeSearch();
+  }
+  componentDidMount() {
+    // The first "newtab" parameter here is called the "healthReportKey" and needs
+    // to be "newtab" so that BrowserUsageTelemetry.jsm knows to handle events with
+    // this name, and can add the appropriate telemetry probes for search. Without the
+    // correct name, certain tests like browser_UsageTelemetry_content.js will fail (See
+    // github ticket #2348 for more details)
```

Comment:
This comment block should go next to the `new ContentSearchUIController` in `initializeSearch`